### PR TITLE
Adding information for NETINFO environmental variable

### PR DESCRIPTION
**Pull request by gizas**
Adding documentation information for NETINFO envrionmental variable introdused in this PR: https://github.com/elastic/beats/pull/36506

- Relates https://github.com/elastic/integrations/issues/6674

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 1]
```diff
@@ -560,4 +560,20 @@ contains your CA's certificate.
 
 // end::kibana-ca[]
 
+// tag::netinfo[]
+|
+[id="env-{type}-netinfo"]
+`NETINFO`
+
+| (bool) When `false`, disables `netinfo.enabled` parameter of `add_host_metadata` processor.
+Setting this to `false` is recommended for large scale setups where the host.ip and host.mac fields index size increases.
```
Comment (ruflin) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:569]: Lets add some more details here around what the problem is -> tons of ip addresses, tons of storage needed, slower queries

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 2]
```diff
@@ -560,4 +560,20 @@ contains your CA's certificate.
 
 // end::kibana-ca[]
 
+// tag::netinfo[]
+|
+[id="env-{type}-netinfo"]
+`NETINFO`
```
Comment (ruflin) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:566]: Seems this changed in the PR to ELASTIC_NETINFO?

File: docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc [hunk 3]
```diff
@@ -145,9 +145,21 @@ Policy configuration of {k8s} package can heavily affect the amount of metrics c
...
+
+https://github.com/elastic/integrations/blob/main/docs/dashboard_guidelines.md[Dashboard Guidelines] document can provide guidance on how to implement your dashboards and is constantly updated also to track needs of observability at scale.
+
 User experience regarding Dashboard responses, is also affected from the size of data being requested. As dashboards can contain multiple visualisations, the general consideration is to split visualisations and group them according to the frequency of access. The less number of visualisations tends to improve user experience.
 
-Additionally, https://github.com/elastic/integrations/blob/main/docs/dashboard_guidelines.md[Dashboard Guidelines] is constantly updated also to track needs of observability at scale.
+[discrete]
+=== Disabling indexing host.ip and host.mac fields 
+
+A new environemntal variable `NETINFO: false` has been introduced to globally disable the indexing of fields `host.ip` and `host.mac` in your {k8s} integration. For more information see <<agent-environment-variables>>.
```
Comment (ruflin) [inline docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc:159]: Does it disable indexing all `host.ip` fields or just reduces the number?

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc (hunk 1 above)
Comment (gizas) [reply docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:569]: Done. Let me know if is ok

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc (hunk 2 above)
Comment (gizas) [reply docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc]: Updated. Was in the middle of the changes 👍

File: docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc (hunk 3 above)
Comment (gizas) [reply docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc]: Disables them

**Comment by gizas** [comment]
Set default value ELASTIC_NETINFO: false as per discussion here https://github.com/elastic/beats/pull/36506#issuecomment-1709769966

File: docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc [hunk 4]
```diff
@@ -145,9 +145,21 @@ Policy configuration of {k8s} package can heavily affect the amount of metrics c
...
+
 User experience regarding Dashboard responses, is also affected from the size of data being requested. As dashboards can contain multiple visualisations, the general consideration is to split visualisations and group them according to the frequency of access. The less number of visualisations tends to improve user experience.
 
-Additionally, https://github.com/elastic/integrations/blob/main/docs/dashboard_guidelines.md[Dashboard Guidelines] is constantly updated also to track needs of observability at scale.
+[discrete]
+=== Disabling indexing host.ip and host.mac fields 
+
+A new environemntal variable `ELASTIC_NETINFO: false` has been introduced to globally disable the indexing of `host.ip` and `host.mac` fields in your {k8s} integration. For more information see <<agent-environment-variables>>.
+
+Setting this to `false` is recommended for large scale setups where the host.ip and host.mac fields index size increases. The number of ips and mac addresses reported, is increasing significantly as kubenetes cluster is growing. This leads to increase indexing time considerably, extra storage is needed and additional overhead in visualisation rendering required.
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc:161]:
```suggestion
Setting this to `false` is recommended for large scale setups where the `host.ip` and `host.mac` fields' index size increases. The number of IPs and MAC addresses reported increases significantly as a Kubenetes cluster grows. This leads to considerably increased indexing time, as well as the need for extra storage and additional overhead for visualization rendering.
```

File: docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc [hunk 5]
```diff
@@ -145,9 +145,21 @@ Policy configuration of {k8s} package can heavily affect the amount of metrics c
 - Disable events dataset
 - Disable {k8s} control plane datasets in Cloud managed {k8s} instances (see more info ** <<running-on-gke-managed-by-fleet>>, <<running-on-eks-managed-by-fleet>>, <<running-on-aks-managed-by-fleet>> pages)
 
+
+[discrete]
+=== Dashboards and Visualisations 
+
+https://github.com/elastic/integrations/blob/main/docs/dashboard_guidelines.md[Dashboard Guidelines] document can provide guidance on how to implement your dashboards and is constantly updated also to track needs of observability at scale.
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc:152]:
```suggestion
The https://github.com/elastic/integrations/blob/main/docs/dashboard_guidelines.md[Dashboard Guidelines] document provides guidance on how to implement your dashboards and is constantly updated to track the needs of Observability at scale.
```

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc [hunk 6]
```diff
@@ -560,4 +560,20 @@ contains your CA's certificate.
...
 
+// tag::elastic-netinfo[]
+|
+[id="env-{type}-elastic-netinfo"]
+`ELASTIC_NETINFO`
+
+| (bool) When `false`, disables `netinfo.enabled` parameter of `add_host_metadata` processor.
+Setting this to `false` is recommended for large scale setups where the host.ip and host.mac fields index size increases.
+
+By default, {agent} initialises the add_host_metadata processor with netinfo.enabled parameter enabled. This parameter includes IP addresses and MAC addresses as fields host.ip and host.mac.
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:571]:
```suggestion
By default, {agent} initializes the `add_host_metadata` processor with the `netinfo.enabled` parameter enabled. This parameter includes IP addresses and MAC addresses as fields `host.ip` and `host.mac`. 
```

**Review by kilfoyle — COMMENTED** [review]
Looks great! Just a few suggestions for the text.

**Comment by gizas** [comment]
Any other suggestions or please can I have an approval here?

**Review by ChrsMark — APPROVED** [review]

File: docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc (hunk 1 above)
Comment (ruflin) [reply docs/en/ingest-management/elastic-agent/configuration/env/shared-env.asciidoc:569]: I think we can still add more details here but lets do this as a follow up.

**Pull request merged**